packages/cdktf/lib/terraform-iterator.ts

```diff
@@ -222,6 +251,88 @@ export abstract class TerraformIterator implements ITerraformIterator {
       })
     );
   }
+
+  /**
+   * Creates a for expression that maps the iterators to its keys.
+   * For lists these would be the indices, for maps the keys.
+   * As this returns an IResolvable you might need to wrap the output in
+   * a Token, e.g. `Token.asString`.
+   */
+  public keys(): IResolvable {
+    return Token.asAny(
+      forExpression(this._getForEachExpression(), FOR_EXPRESSION_KEY)
+    );
+  }
+
+  /**
+   * Creates a for expression that maps the iterators to its value in case it is a map.
+   * For lists these would stay the same.
+   * As this returns an IResolvable you might need to wrap the output in
+   * a Token, e.g. `Token.asString`.
+   */
+  public values(): IResolvable {
+    return Token.asAny(
+      forExpression(this._getForEachExpression(), FOR_EXPRESSION_VALUE)
+    );
+  }
+
+  /**
+   * Creates a for expression that accesses the key on each element of the iterator.
+   * As this returns an IResolvable you might need to wrap the output in
+   * a Token, e.g. `Token.asString`.
+   * @param property The property of the iterators values to map to
+   */
+  public pluckProperty(property: string): IResolvable {
+    return Token.asAny(
+      forExpression(
+        this._getForEachExpression(),
+        propertyAccess(FOR_EXPRESSION_VALUE, [property])
+      )
+    );
+  }
+
+  /**
+   * Creates a for expression that results in a list.
+   * This method allows you to create every possible for expression, but requires more knowledge about
+   * Terraforms for expression syntax.
+   * For the most common use cases you can use keys(), values(), and pluckProperty() instead.
+   *
+   * You may write any valid Terraform for each expression, it will result
+   * in `[ for key, val in var.myIteratorSource: <expression> ]`.
+   *
+   * As this returns an IResolvable you might need to wrap the output in
+   * a Token, e.g. `Token.asString`.
```

Comment:
Would be nice to accompany the above terraform example with a CDKTF call showing how to use this API